app/src/main/java/org/wikipedia/readinglist/database/ReadingListDbHelper.java

```diff
@@ -391,6 +391,27 @@ public void updatePage(@NonNull ReadingListPage page) {
         }
     }
 
+    public void updateMetadataByTitle(@NonNull String title, @NonNull String lang,
+                                      @Nullable String description, @Nullable String thumbUrl) {
+        SQLiteDatabase db = getWritableDatabase();
+        db.beginTransaction();
+        try {
+            ContentValues contentValues = new ContentValues();
+            contentValues.put(ReadingListPageContract.Col.THUMBNAIL_URL.getName(), thumbUrl);
+            contentValues.put(ReadingListPageContract.Col.DESCRIPTION.getName(), description);
+            int result = db.update(ReadingListPageContract.TABLE, contentValues,
+                    ReadingListPageContract.Col.API_TITLE.getName() + " = ? AND "
```

Comment:
Would it be possible to also check `ReadingListPageContract.Col.DISPLAY_TITLE.getName()` to see if the title matches? Please see `findPageInAnyList()` and `getPageByTitle()`.

Reply:
Why is that necessary?  (When would there be multiple pages with the same API_TITLE and language, but different DISPLAY_TITLE?)

Reply:
In `ReadingListPageContract`, the `DISPLAY_TITLE` was used to be the "default" title of the data. And the reason why the checks exist is to resolve the language variant issues.

Reply:
alright!